PluginException: Must not execute inside read action

```
com.intellij.diagnostic.PluginException: Must not execute inside read action; If you access or modify model on EDT consider wrapping your code in WriteIntentReadAction  or ReadAction; see https://jb.gg/ij-platform-threading for details
Current thread: Thread[#67,DefaultDispatcher-worker-19,5,main] 1725915426 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[#46,AWT-EventQueue-0,10,main] 829848305 [Plugin: MavenRunHelper]
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImplKt.getProviderIfApplicable(FileEditorProviderManagerImpl.kt:263)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImplKt.access$getProviderIfApplicable(FileEditorProviderManagerImpl.kt:1)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImpl$getProviders$sharedProviders$1$1$1$2.invokeSuspend(FileEditorProviderManagerImpl.kt:146)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImpl$getProviders$sharedProviders$1$1$1$2.invoke(FileEditorProviderManagerImpl.kt)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImpl$getProviders$sharedProviders$1$1$1$2.invoke(FileEditorProviderManagerImpl.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:67)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
	at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
	at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
	at kotlinx.coroutines.TimeoutKt.withTimeout-KLykuaI(Timeout.kt:72)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImpl$getProviders$sharedProviders$1$1$1.invokeSuspend(FileEditorProviderManagerImpl.kt:140)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:610)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runDefaultDispatcherTask(CoroutineScheduler.kt:1194)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:906)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:775)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:762)
Caused by: com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Must not execute inside read action; If you access or modify model on EDT consider wrapping your code in WriteIntentReadAction  or ReadAction; see https://jb.gg/ij-platform-threading for details
Current thread: Thread[#67,DefaultDispatcher-worker-19,5,main] 1725915426 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[#46,AWT-EventQueue-0,10,main] 829848305
	at com.intellij.util.concurrency.ThreadingAssertions.createThreadAccessException(ThreadingAssertions.java:258)
	at com.intellij.util.concurrency.ThreadingAssertions.throwThreadAccessException(ThreadingAssertions.java:245)
	at com.intellij.util.concurrency.ThreadingAssertions.assertNoReadAccess(ThreadingAssertions.java:177)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.doInitTree(MavenProjectsManager.java:279)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.initProjectsTree(MavenProjectsManager.java:269)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.getProjectsTree(MavenProjectsManager.java:621)
	at org.jetbrains.idea.maven.project.MavenProjectsManager.findProject(MavenProjectsManager.java:486)
	at krasa.mavenhelper.analyzer.MyFileEditorProvider.isPomFile(MyFileEditorProvider.java:37)
	at krasa.mavenhelper.analyzer.MyFileEditorProvider.accept(MyFileEditorProvider.java:28)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImplKt.checkProvider(FileEditorProviderManagerImpl.kt:272)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImplKt.getProviderIfApplicable$lambda$0(FileEditorProviderManagerImpl.kt:244)
	at com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:111)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable$lambda$0(InternalReadAction.kt:101)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0$0$0(cancellableReadAction.kt:32)
	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:121)
	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:121)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:881)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1232)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0$0(cancellableReadAction.kt:30)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:73)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:152)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0(cancellableReadAction.kt:28)
	at com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
	at com.intellij.concurrency.ThreadContext.installThreadContext$default(threadContext.kt:303)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal(cancellableReadAction.kt:24)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable(InternalReadAction.kt:100)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadAction(InternalReadAction.kt:86)
	at com.intellij.openapi.application.rw.InternalReadAction.readLoop(InternalReadAction.kt:73)
	at com.intellij.openapi.application.rw.InternalReadAction.access$readLoop(InternalReadAction.kt:15)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invokeSuspend(InternalReadAction.kt:52)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invoke(InternalReadAction.kt)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invoke(InternalReadAction.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:67)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:43)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:165)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:50)
	at com.intellij.openapi.application.rw.PlatformReadWriteActionSupport.executeReadAction(PlatformReadWriteActionSupport.kt:66)
	at com.intellij.openapi.application.ReadWriteActionSupport.executeReadAction$default(ReadWriteActionSupport.kt:15)
	at com.intellij.openapi.application.CoroutinesKt.constrainedReadAction(coroutines.kt:60)
	at com.intellij.openapi.application.CoroutinesKt.readAction(coroutines.kt:27)
	at com.intellij.openapi.fileEditor.impl.FileEditorProviderManagerImplKt.getProviderIfApplicable(FileEditorProviderManagerImpl.kt:243)
	... 17 more

```

--- a/src/main/java/krasa/mavenhelper/analyzer/MyFileEditorProvider.java
+++ b/src/main/java/krasa/mavenhelper/analyzer/MyFileEditorProvider.java
@@ -34,7 +34,10 @@ public class MyFileEditorProvider implements FileEditorProvider, DumbAware {
 		if (!isPotentialPomFile(name)) return false;
 
 		MavenProjectsManager instance = MavenProjectsManager.getInstance(project);
-		final MavenProject mavenProject = instance == null ? null : instance.findProject(file);
+		// Guard against calling findProject() before the project tree is initialized, as that triggers
+		// doInitTree() which asserts no read access and throws a PluginException when called from a read action.
+		if (instance == null || !instance.isMavenizedProject()) return false;
+		final MavenProject mavenProject = instance.findProject(file);
 		if (mavenProject != null) {
 			return mavenProject.getFile().equals(file);
 		}
@@ -52,7 +55,9 @@ public class MyFileEditorProvider implements FileEditorProvider, DumbAware {
 //		https://github.com/krasa/MavenHelper/issues/130
 //		LOG.assertTrue(accept(project, file));
 
-		return new UIFormEditor(project, file);
+		MavenProjectsManager instance = MavenProjectsManager.getInstance(project);
+		MavenProject mavenProject = (instance != null && instance.isMavenizedProject()) ? instance.findProject(file) : null;
+		return new UIFormEditor(project, file, mavenProject);
 	}
 
 	@Override

--- a/src/main/java/krasa/mavenhelper/analyzer/MyHighlightingTree.java
+++ b/src/main/java/krasa/mavenhelper/analyzer/MyHighlightingTree.java
@@ -41,10 +41,10 @@ public class MyHighlightingTree extends Tree implements DataProvider {
 		if (object instanceof MyTreeUserObject) {
 			if (((MyTreeUserObject) object).isHighlight()) {
 				ApplicationSettings state = MavenHelperApplicationService.getInstance().getState();
-				if (UIUtil.isUnderDarcula()) {
-					return darker(new JBColor(new Color(state.getSearchBackgroundColor()), new Color(state.getSearchBackgroundColor())), 8);
-				} else {
+				if (JBColor.isBright()) {
 					return softer(new JBColor(new Color(state.getSearchBackgroundColor()), new Color(state.getSearchBackgroundColor())));
+				} else {
+					return darker(new JBColor(new Color(state.getSearchBackgroundColor()), new Color(state.getSearchBackgroundColor())), 8);
 				}
 			}
 		}

--- a/src/main/java/krasa/mavenhelper/analyzer/UIFormEditor.java
+++ b/src/main/java/krasa/mavenhelper/analyzer/UIFormEditor.java
@@ -10,8 +10,8 @@ import com.intellij.openapi.project.Project;
 import com.intellij.openapi.util.UserDataHolderBase;
 import com.intellij.openapi.vfs.VirtualFile;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.jetbrains.idea.maven.project.MavenProject;
-import org.jetbrains.idea.maven.project.MavenProjectsManager;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -30,9 +30,8 @@ public final class UIFormEditor extends UserDataHolderBase implements /* Navigat
 	private final VirtualFile file;
 	private GuiForm myEditor;
 
-	public UIFormEditor(@NotNull Project project, final VirtualFile file) {
+	public UIFormEditor(@NotNull Project project, final VirtualFile file, @Nullable MavenProject mavenProject) {
 		this.file = file;
-		final MavenProject mavenProject = MavenProjectsManager.getInstance(project).findProject(file);
 		if (mavenProject != null) {
 			myEditor = new GuiForm(project, file, mavenProject);
 		} else {

